Update bootstrap from V4-Alpha6 to V4.3.1

ShmoogleOsukami opened the pull request:
This is the absolute minimum amount of work required to get bootstrap updated.

There is a lot of classes that'll need updating but as of right now everything looks and functions the same as before.

This will be a good foundation for getting everything cleaned up. We'll be able to do page by page PRs to update the CSS now.

RoboticMind commented:
Looks like you need to update your index.htm because of #177

barton2526 commented on _includes/_head.htm:
```diff
@@ -8,13 +8,12 @@
-	<!--     Fonts and icons     -->
-	<link href="https://fonts.googleapis.com/css?family=Montserrat:400,700,200" rel="stylesheet" />
+	<!--     Fonts and icons     
+	<link href="https://fonts.googleapis.com/css?family=Montserrat:400,700,200" rel="stylesheet" />-->
```
There is no reason given for this commenting out.  Is this part of the PR?

ShmoogleOsukami replied:
After updating Bootstrap the font changed and messed with the layout of the site. Commenting this out fixed it to a degree.

I should really fix the Font once and for all but we have 3 things messing with it. Bootstrap does stuff to the font, NowUI messes with what bootstrap does and then we manually define a font as well. 

I'd sooner just use bootstraps default font or one of the fonts it has with it.

I don't know much about setting up fonts for websites. I should look into it.

ShmoogleOsukami replied:
Hang on, I got a fix for this, I can reinstate the font, Just have to fix a few oddly sized bits of text on the index page.

ShmoogleOsukami replied:
This is with the font,
![screencapture-127-0-0-1-4000-2019-05-20-15_25_59](https://user-images.githubusercontent.com/772631/58029545-b94e7580-7b14-11e9-9c89-975c28bd1614.png)

This is without the font. It looks better without but ideally, I'd like to remove Now-ui so bootstraps responsive font stuff stops getting overridden.

![Uploading screencapture-127-0-0-1-4000-index-htm-2019-05-20-15_33_31.png…]()

barton2526 replied:
Recommend a new PR to revisit the font issue separately.  Then I could just give the current PR the okay.  I think the font discussion should be a separate one.

barton2526 commented on _layouts/layout.htm:
```diff
@@ -7,11 +7,9 @@
 
 	<!--   Core JS Files   -->
-	<script src="/assets/js/core/jquery.3.2.1.min.js" type="text/javascript"></script>
+	<script src="/assets/js/core/jquery-3.4.1.min.js" type="text/javascript"></script>
```
Note: JQuery is also being updated.  Does this affect/break anything?

ShmoogleOsukami replied:
As far as my local tests goes, Everything seems to function as before. Bootstrap isn't reporting any problems nor is NowUI. Everything looks correct.

barton2526 commented:
Approved

barton2526 commented:
Waiting for travis

barton2526 commented:
I would like 1 more review before we merge

cyrossignol commented on index.htm:
```diff
@@ -368,11 +289,11 @@ <h4>Alternative Linux sources</h4>
-                    <a class="btn btn-round grcbtn" href="https://boinc.berkeley.edu/download_all.php">BOINC Clients</a><br/>
-                    <a class="btn btn-round grcbtn" href="https://boinc.berkeley.edu/wiki/Add-on_software">BOINC Add-on software</a><br/>
-                    <a class="btn btn-round grcbtn" href="https://github.com/marius311/boinc-server-docker">BOINC server Docker container</a>
+                    <a class="btn rounded-pill grcbtn" href="https://boinc.berkeley.edu/download_all.php">BOINC Clients</a><br/>
```
The other merged `btn-round` download links came back with the NOW-UI class. Could use `rounded-pill` as well so the font weight matches--that class will go away anyway.

ShmoogleOsukami commented on index.htm:
```diff
@@ -6,52 +6,52 @@
-                <h4 class="text-bold">Rewarding Volunteer Distributed Computing</h4>
+            <div class="col-12 text-center">
+                <img class="img-fluid" src="/assets/img/GridcoinBlocks_Header_Transparent.png" alt="Gridcoin Logo">
+                <h1 class="text-weight-bold">Rewarding Volunteer Distributed Computing</h1>
```
Opted for keeping it a H1 tag but styling as a H2 should help with the mobile visuals greatly.

cyrossignol replied:
Great 👍

ShmoogleOsukami commented:
I've updated all the pill buttons to the correct class and went ahead and removed most of the CSS in gridcoin.css for the index page and replaced them with bootstrap classes.
Layout is largely the same just minor spacing differences.

cyrossignol commented on index.htm:
```diff
@@ -6,52 +6,52 @@
-                <h4 class="text-bold">Rewarding Volunteer Distributed Computing</h4>
+            <div class="col-12 text-center">
+                <img class="img-fluid" src="/assets/img/GridcoinBlocks_Header_Transparent.png" alt="Gridcoin Logo">
+                <h1 class="h2 pt-3 text-weight-bold">Rewarding Volunteer Distributed Computing</h1>
```
Didn't you want `font-weight-bold` here?

ShmoogleOsukami replied:
oh nice catch yeah. Will fix.

cyrossignol replied:
Resolved.
